src/main/java/com/google/jenkins/plugins/credentials/oauth/JsonServiceAccountConfig.java

```diff
@@ -53,36 +63,109 @@
   private static final long serialVersionUID = 6818111194672325387L;
   private static final Logger LOGGER =
       Logger.getLogger(JsonServiceAccountConfig.class.getSimpleName());
-  private String jsonKeyFile;
+  @CheckForNull
+  private String filename;
+  @CheckForNull
+  private SecretBytes secretJsonKey;
+  @Deprecated   // for migration purpose
+  @CheckForNull
+  private transient String jsonKeyFile;
   private transient JsonKey jsonKey;
 
+  /** @since 0.8 */
   @DataBoundConstructor
-  public JsonServiceAccountConfig(FileItem jsonKeyFile,
-      String prevJsonKeyFile) {
-    if (jsonKeyFile != null && jsonKeyFile.getSize() > 0) {
+  public JsonServiceAccountConfig() {}
+
+  /**
+   * For being able to load credentials created with versions < 0.8
+   * and backwards compatibility with external callers.
+   *
+   * @param jsonKeyFile The uploaded json key file
+   * @param prevJsonKeyFile The path of the previous json key file
+   * @since 0.3
+   */
+  @Deprecated
+  public JsonServiceAccountConfig(
+      FileItem jsonKeyFile, String prevJsonKeyFile) {
+    this.setJsonKeyFileUpload(jsonKeyFile);
+    if (filename == null && prevJsonKeyFile != null) {
+      this.filename = extractFilename(prevJsonKeyFile);
+      this.secretJsonKey = getSecretBytesFromFile(prevJsonKeyFile);
+    }
+  }
+
+  /**@param jsonKeyFileUpload uploaded json key file */
```

Comment:
This Javadoc header is improperly formatted and is missing a method description. Additionally capitalization and punctuation need correcting.

Reply:
I think for the most part getters and setters are self documenting and the parameter description clears describes what is being set. I'm struggling to come up with a description to add here that is both not redundant and not exposing unnecessary details about the implementation. What are your thoughts?